### PR TITLE
Add OpenAPI spec and generated client

### PR DESCRIPTION
**Pull request by vish-mv**
### Purpose
Integrate the new Twitter (X) OpenAPI specification (Twitter API V2) by generating the client, types, and utilities for the updated API.

### Changes

- Added the new Twitter (X) OpenAPI spec (Twitter API V2) to the project.
- Generated client code based on the new API specification.
- Generated type definitions for the updated API endpoints.
- Generated utility functions to support the new client and type definitions.

### CODE REVIEW

**Comment by NipunaRanasinghe** [comment]
we also need to add a `sanitations.md` file including all the sanitations we did on the original spec and, the exact command we used to generate the client using the openAPI tool. (A example PR with the expected file structures can be found in https://github.com/ballerina-platform/module-ballerinax-asana/pull/2/files)

File: docs/spec/twitter_open_api_spec.json
[the remark this answers is not in the document]
Comment (vish-mv) [reply docs/spec/twitter_open_api_spec.json:1]: Updated.

File: docs/spec/sanitations.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+_Authors_: vish-mv , @loshan20011  \
```
Comment (NipunaRanasinghe) [inline docs/spec/sanitations.md:1]:
```suggestion
_Authors_: @vish-mv @loshan20011  \
```

**Comment by NipunaRanasinghe** [comment]
@vish-mv lets add the missing `docs/license.txt` file.

File: ballerina/client.bal [hunk 2]
```diff
@@ -13,3 +16,676 @@
...
+            if config.proxy is http:ProxyConfig {
+                httpClientConfig.proxy = check config.proxy.ensureType(http:ProxyConfig);
+            }
+        }
+        http:Client httpEp = check new (serviceUrl, httpClientConfig);
+        self.clientEp = httpEp;
+        return;
+    }
+
+    resource isolated function delete '2/dm_events/[DmEventId event_id](map<string|string[]> headers = {}) returns DeleteDmResponse|error {
```
Comment (NipunaRanasinghe) [inline ballerina/client.bal:58]: It seems like all the resource functions in the generated client is having the API version (`2/`) as a prefix, which is redundant and also impacts the readability and usability. 

Therefore we have the option to sanitize the OpenAPI contract by 
1. making `https://developer.twitter.com/2/` as the base url and
2. removing the `/2` prefix from all the paths in the contract

So shall we try that out and see if it works fine?

```suggestion
    resource isolated function delete 'dm_events/[DmEventId event_id](map<string|string[]> headers = {}) returns DeleteDmResponse|error {
```

Comment (vish-mv) [reply ballerina/client.bal]: Solved.

File: docs/spec/sanitations.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
+_Authors_: @vish-mv @loshan20011  \
+_Created_: 2024/03/26 \
+_Updated_: 2024/03/26 \
+_Edition_: Swan Lake
+
+# Sanitation for OpenAPI specification
+
+This document records the sanitation done on top of the official OpenAPI specification from Twitter (X). The OpenAPI specification is obtained from the [Twitter OpenAPI Documentation](https://api.twitter.com/2/openapi.json/). These changes are implemented to enhance the overall usability and readability of the generated client.
+
+## Base URL Changes
```
Comment (NipunaRanasinghe) [inline docs/spec/sanitations.md:10]:
```suggestion
```

File: docs/spec/sanitations.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+1. **Change the `url` property of the `servers` object**:
+   - **Original**: `https://api.twitter.com`
+   - **Updated**: `https://api.twitter.com/2`
+   - **Reason**: This change is made to ensure that all API paths are relative to the versioned base URL (`/2`), which improves the consistency and usability of the APIs.
+
+## Function Name Changes
```
Comment (NipunaRanasinghe) [inline docs/spec/sanitations.md:17]:
```suggestion
```

**Review by NipunaRanasinghe — APPROVED** [review]

**Pull request merged**